src/cascadia/TerminalConnection/TelnetConnection.cpp

```diff
@@ -0,0 +1,265 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "pch.h"
+#include "TelnetConnection.h"
+#include <LibraryResources.h>
+
+#include "TelnetConnection.g.cpp"
+
+#include "../../types/inc/Utils.hpp"
+
+using namespace ::Microsoft::Console;
+
+namespace winrt::Microsoft::Terminal::TerminalConnection::implementation
+{
+    TelnetConnection::TelnetConnection(const hstring& hostname) :
+        _reader{ nullptr },
+        _hostname{ hostname }
+    {
+    }
+
+    // Method description:
+    // - ascribes to the ITerminalConnection interface
+    // - creates the output thread (where we will do the authentication and actually connect to Azure)
+    void TelnetConnection::Start()
+    {
+        // Create our own output handling thread
+        // Each connection needs to make sure to drain the output from its backing host.
+        _hOutputThread.reset(CreateThread(nullptr,
+                                          0,
+                                          StaticOutputThreadProc,
+                                          this,
+                                          0,
+                                          nullptr));
+
+        THROW_LAST_ERROR_IF_NULL(_hOutputThread);

```

Comment:
This should maybe force a transition into `Failed` using a function-try like ConptyConnection